### ISSUE TITLE
NullPointerException in DubProjectImportBuilder

### ISSUE THREAD

**Issue by SingingBush** [intellij-dlanguage/intellij-dlanguage#872]
release 1.31 on Intellij IDEA:

```
java.lang.NullPointerException: Parameter specified as non-null is null: method io.github.intellij.dub.project.DubProjectImportBuilder.commit, parameter modifiableArtifactModel
    at io.github.intellij.dub.project.DubProjectImportBuilder.commit(DubProjectImportBuilder.kt)
    at com.intellij.projectImport.ProjectImportBuilder.commit(ProjectImportBuilder.java:50)
    at com.intellij.ide.impl.NewProjectUtil.doCreate(NewProjectUtil.kt:155)
    at com.intellij.ide.impl.NewProjectUtil.createFromWizard(NewProjectUtil.kt:84)
    at com.intellij.ide.impl.NewProjectUtil.createFromWizard$default(NewProjectUtil.kt:82)
    at com.intellij.ide.actions.ImportModuleActionKt.doCreateFromWizard(ImportModuleAction.kt:199)
    at com.intellij.ide.actions.ImportModuleActionKt.access$doCreateFromWizard(ImportModuleAction.kt:1)
    at com.intellij.ide.actions.ImportModuleAction$Companion.doImport(ImportModuleAction.kt:55)
    at com.intellij.ide.actions.ImportModuleAction$Companion.doImport(ImportModuleAction.kt:43)
    at com.intellij.ide.actions.ImportModuleAction.doImport(ImportModuleAction.kt)
    at com.intellij.ide.actions.ImportProjectAction.actionPerformed(ImportProjectAction.java:13)
    at com.intellij.openapi.actionSystem.ex.ActionUtil.doPerformActionOrShowPopup(ActionUtil.java:333)
    at com.intellij.openapi.actionSystem.ex.ActionUtil.lambda$performActionDumbAwareWithCallbacks$4(ActionUtil.java:307)
    at com.intellij.openapi.actionSystem.ex.ActionUtil.performDumbAwareWithCallbacks(ActionUtil.java:356)
    at com.intellij.openapi.actionSystem.ex.ActionUtil.performActionDumbAwareWithCallbacks(ActionUtil.java:307)
    at com.intellij.openapi.actionSystem.impl.ActionMenuItem.lambda$performAction$5(ActionMenuItem.java:299)
    at com.intellij.openapi.wm.impl.FocusManagerImpl.runOnOwnContext(FocusManagerImpl.java:225)
    at com.intellij.openapi.actionSystem.impl.ActionMenuItem.performAction(ActionMenuItem.java:292)
    at com.intellij.openapi.actionSystem.impl.ActionMenuItem.lambda$new$1(ActionMenuItem.java:81)
    at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:113)
    at com.intellij.openapi.application.TransactionGuardImpl.performUserActivity(TransactionGuardImpl.java:94)
    at com.intellij.openapi.actionSystem.impl.ActionMenuItem.lambda$new$2(ActionMenuItem.java:81)
    at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318)
    at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:788)
    at java.awt.EventQueue$3.run(EventQueue.java:739)
    at java.awt.EventQueue$3.run(EventQueue.java:731)
    at java.security.AccessController.doPrivileged(AccessController.java:399)
    at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
    at java.awt.EventQueue.dispatchEvent(EventQueue.java:758)
```

**Comment by rikkimax** [comment]
Looks like a missing ``?`` on the function parameter. The body appears to already handle a null case.

So not something I did which is good :)

**Comment by SingingBush** [comment]
annoyingly the nullability of parameters sometimes changes between ide versions and the Kotlin classes need to be specific about it. This is one of the reasons that the plugin doesn't support too wide a range of IDE releases.

**Comment by rikkimax** [comment]
I think this kinda solidifies my view that the Kotlin stuff should be rewritten into Java.

It's not just a pain to look for things but also is a genuine issue.

**Comment by SingingBush** [comment]
It's probably just teething issues while Jetbrains migrate their codebase from Java to Kotlin. Their platform has evolved quite a bit in the last few years. Fortunately documentation is pretty good now, certainly a lot better than when I got involved with the project.

From what I've seen, Java devs pick up Kotlin quickly and after a while prefer to stick with it. I've seen numerous employers in the city where I work transition to Kotlin and any new Jetbrains plugins or Android apps should probably be using it. I've also used it for Spring Boot apps and Apache Flink jobs and have found that benefits outweigh the cons.
